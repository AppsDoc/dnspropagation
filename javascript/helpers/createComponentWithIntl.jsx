/**
 * @see https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl#jest
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

const createComponentWithIntl = (children, props = { locale: 'en' }) =>
  renderer.create(
    <IntlProvider {...props}>
      {children}
    </IntlProvider>
  );

export default createComponentWithIntl;
