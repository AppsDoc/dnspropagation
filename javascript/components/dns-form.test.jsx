/*
 * The MIT License (MIT)
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
import React from 'react';
import DnsForm from './dns-form';
import createComponentWithIntl from '../helpers/createComponentWithIntl';

const params = {
  domain: 'golang.org',
  working: false,
  type: 'a',
  changeType: jest.fn(),
  changeDomain: jest.fn(),
  onDnsQuerySubmit: jest.fn()
};

test('it should render the component', () => {
  const component = createComponentWithIntl(
    <DnsForm
      submit={params.onDnsQuerySubmit}
      working={params.working}
      type={params.type}
      domain={params.domain}
      changeType={params.handleTypeChange}
      changeDomain={params.handleDomainChange}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
