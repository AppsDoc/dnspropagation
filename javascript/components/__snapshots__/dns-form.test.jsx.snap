// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the component 1`] = `
<form
  action="/api/v1/query"
  method="post"
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="col-lg-4"
    >
      <input
        className="form-control"
        id="domain"
        onChange={[Function]}
        placeholder="What is the domain you want to check?"
        required={true}
        type="text"
        value="golang.org"
      />
    </div>
    <div
      className="col-lg-2"
    >
      <select
        className="form-control"
        onChange={[Function]}
        required={true}
        value="a"
      >
        <option
          value="a"
        >
          A
        </option>
        <option
          value="aaaa"
        >
          AAAA
        </option>
        <option
          value="caa"
        >
          CAA
        </option>
        <option
          value="cname"
        >
          CNAME
        </option>
        <option
          value="mx"
        >
          MX
        </option>
        <option
          value="ns"
        >
          NS
        </option>
        <option
          value="ptr"
        >
          PTR
        </option>
        <option
          value="soa"
        >
          SOA
        </option>
        <option
          value="srv"
        >
          SRV
        </option>
        <option
          value="txt"
        >
          TXT
        </option>
      </select>
    </div>
    <div
      className="col-lg-2"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        <span
          className="glyphicon glyphicon-search"
        >
          Â 
        </span>
        <span>
          <span>
            Query a record for golang.org
          </span>
        </span>
      </button>
    </div>
  </div>
</form>
`;
